### PR TITLE
Namespace filter for service discovery (issue #1000). 

### PR DESCRIPTION
**Pull request by mbialkowski1**

### CODE REVIEW

File: spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-fabric8-client-discovery/src/test/java/org/springframework/cloud/kubernetes/fabric8/configmap/Fabric8DiscoveryNamespaceFilterIT.java [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+import org.springframework.cloud.kubernetes.integration.tests.commons.K8SUtils;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.web.reactive.function.client.WebClient;
+
+/**
+ * @author mbialkowski1
+ */
+class Fabric8DiscoveryNamespaceFilterIT {
```
Comment (wind57) [inline spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-fabric8-client-discovery/src/test/java/org/springframework/cloud/kubernetes/fabric8/configmap/Fabric8DiscoveryNamespaceFilterIT.java:54]: this is fabulous! I'd like to ask for an integration test in k8s-client also, the reason for that is that eventually we will move away from fabric8 for discovery (just like we did for reload for example). again, this is awesome and thank you!

**Comment by ryanjbaxter** [comment]
Done!

**Comment by wind57** [comment]
I'l leaving a note here for myself mainly, we need this change in `KubernetesCatalogWatch` too. The fabric8 one + when we add a k8s-native implementation of catalog watcher, too.

**Comment by mbialkowski1** [comment]
> @mbialkowski1 tell us when this is done and I will gladly take a look (most probably Ryan too) thank you

I think I'm done, I added an integration test for `org.springframework.cloud.kubernetes.discovery.KubernetesDiscoveryClient`.
I also found the problem in the method `org.springframework.cloud.kubernetes.client.discovery.KubernetesInformerDiscoveryClient#getInstances` 
We use only one service:
```
V1Service service = properties.allNamespaces() ? this.serviceLister.list().stream()
				.filter(svc -> serviceId.equals(svc.getMetadata().getName())).findFirst().orElse(null)
```

If a service with the same name is in several different namespaces then we lost information about it.
I corrected it, can you see if it is ok? (https://github.com/spring-cloud/spring-cloud-kubernetes/pull/1113/commits/fed3e4f74a581ad1455096a61849e76ba9974c26)
What else should I do/improve in this PR?

**Comment by wind57** [comment]
I will take a look at this in the upcoming weeks, I just want to work (and clarify some things first), if you do not mind.

**Review by ryanjbaxter — COMMENTED** [review]
Do we still retain the functionality that is `all-namespace` is false and no specific namespace is specified we still use the current namespace?

**Comment by mbialkowski1** [comment]
> Do we still retain the functionality that is `all-namespace` is false and no specific namespace is specified we still use the current namespace?

Yes, it should work like that.

**Pull request merged**